docs/reference/terraform.md

```diff
@@ -0,0 +1,299 @@
+
+### `modules.tf`
+
+!!! abstract Source
+
+    [`terraform/modules.tf`](https://github.com/Office-of-Digital-Services/cdt-ods-disaster-recovery/tree/main/terraform/modules.tf)
+
+This file acts as the central nervous system for the infrastructure, connecting all the individual modules defined in the `/terraform/modules` directory.
+
+- **Orchestration**: It instantiates each core module (`network`, `monitoring`, `database`, etc.).
+- **Wiring**: It passes the outputs from one module as inputs to another. This creates a dependency graph and ensures resources are created in the correct order. For example, the network module's subnet IDs are passed to the application and database modules.
+- **Consistent Naming**: It defines a `locals` block that establishes a consistent naming convention for resources across all modules, based on the environment.
+
+Here is an example of how `modules.tf` wires the `monitoring` and `network` modules together:
+
+```terraform
+module "monitoring" {
+  source                        = "./modules/monitoring"
+  # ... other variables
+}
+
+module "network" {
+  source                     = "./modules/network"
+  log_analytics_workspace_id = module.monitoring.log_analytics_workspace_id
+  # ... other variables
+}
+```
+
+### `security.tf`
+
+!!! abstract Source
+
+    [`terraform/security.tf`](https://github.com/Office-of-Digital-Services/cdt-ods-disaster-recovery/tree/main/terraform/security.tf)
+
+This file centralizes security-related configurations that span across multiple modules. This separation of concerns makes it easier to manage and audit security settings. Its responsibilities include:
+
+- **Key Vault Access Policies**: It creates `azurerm_key_vault_access_policy` resources to grant the managed identities of the application's container apps (`web`, `worker`, `functions`) the necessary permissions to read secrets from Key Vault.
+- **Network Security Group (NSG) Rules**: It defines the specific `azurerm_network_security_rule` resources that allow or deny traffic between the different subnets. For example, it contains rules to allow the application subnet to communicate with the database subnet on the correct port.
+
+### `alerts.tf`
+
+!!! abstract Source
+
+    [`terraform/alerts.tf`](https://github.com/Office-of-Digital-Services/cdt-ods-disaster-recovery/tree/main/terraform/alerts.tf)
+
+This file is dedicated to defining monitoring alerts for the application. It creates an `azurerm_monitor_scheduled_query_rules_alert_v2` resource that:
+
+- Runs a query against Application Insights logs on a schedule (e.g., every 5 minutes).
+- The query checks for exceptions or high-severity traces.
+- If the query returns any results, it triggers an alert that notifies the team via the action group configured in the `monitoring` module.
+
+## Core modules (`terraform/modules/*`)
```

Comment:
Just a minor note, we could put the core modules in alphabetical order in this section to match the "Directory structure" figure above.

Reply:
Surely can do that. The current order is more reflective of the infrastructure / dependency order, e.g. we need the network to exist before we can create a database, and that has to exist before we can create the apps. 

But I am happy to reorder alphabetically so there is no confusion, that seems more straightforward (and easier to change in the future).

Reply:
Added some language in [`a32176f`](https://github.com/Office-of-Digital-Services/cdt-ods-disaster-recovery/pull/465/commits/a32176f9b22e023e689026acf4dd25afe24c9103), LMK what you think.